src/Sentry/Internal/Http/LazyHttpTransport.cs

```diff
@@ -0,0 +1,22 @@
+﻿using Sentry.Extensibility;
+using Sentry.Protocol.Envelopes;
+
+namespace Sentry.Internal.Http;
+
+internal class LazyHttpTransport : ITransport
+{
+    private readonly SentryOptions _options;
+    private HttpTransport? _httpTransport;
+
+    public LazyHttpTransport(SentryOptions options)
+    {
+        _options = options;
+    }
+
+    public Task SendEnvelopeAsync(Envelope envelope, CancellationToken cancellationToken = default)
+    {
+        _httpTransport ??= new HttpTransport(_options, (_options.SentryHttpClientFactory ?? new DefaultSentryHttpClientFactory()).Create(_options));
```

Comment:
How about moving this all under a `Lazy<>` and access it via `_lazyTransport.Value` ?

Reply:
Sure, there's not enough `Lazy` in my PR 😅